cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -355,4 +360,14 @@ public void execute(TxRunnable runnable) throws TransactionFailureException {
   public void execute(int timeoutInSeconds, TxRunnable runnable) throws TransactionFailureException {
     transactional.execute(timeoutInSeconds, runnable);
   }
+
+  @Override
+  public DataTracer getDataTracer(String dataTracerName) {
+    return dataTracerFactory.getTracer(dataTracerName, program.getId().getParent());
+  }
+
+  @Inject(optional = true)
+  private void setDataTracerFactory(@Named("DataTracerFactory") DataTracerFactory dataTracerFactory) {
```

Comment:
Do not quite understand the concept of optional and method injection. Just followed the examples. If we use method injection, do we need to call this method or the variable will be injected without calling?